### DISCUSSION TITLE
Packaging for Debian

### DISCUSSION THREAD

**Discussion by legoktm** [jojo2357/kiwix-zim-updater#30] [Ideas]
Hi, thanks for writing this! I currently maintain the Kiwix/openZIM stack in Debian and intend to package this as well (https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1051782).

The main question I have is regarding the name...typically it would get installed in `/usr/bin/` and things there don't have a `.sh` extension. I am wondering if installing it as `/usr/bin/kiwix-zim-updater` would be a better name and acceptable to you? Or maybe you have a different preference?

**Comment by jojo2357** [comment]
Since this script auto-updates itself, any changes to the name will be a breaking change if they get published to the main branch. 

Also would you like an argparse script? I can cook one of those up real quick or you can pr that ;) .

**Reply by legoktm** [reply]
> any changes to the name will be a breaking change if they get published to the main branch

Ack. I could change the name in just Debian though, would that be fine?

The main issue in Debian context is that we already have packages like "kiwix" and "zim-tools", something named "kiwix-zim" doesn't actually explain what it does, which is provide updates.

> Also would you like an argparse script?

Sorry not sure what you mean, do you mean the Python argparse library?

**Reply by jojo2357** [reply]
> Sorry not sure what you mean, do you mean the Python argparse library?

No, I dunno what its called, but the bash script that autofills the command line args

> Ack. I could change the name in just Debian though, would that be fine?

So long as you abide by the terms of the license and provide the appropriate credit, then I cant really tell you what to do. What do you want *me* to do?

**Reply by legoktm** [reply]
Oh yes, bash completion would be very nice!

> So long as you abide by the terms of the license and provide the appropriate credit, then I cant really tell you what to do. What do you want _me_ to do?

Some times people get very upset when people make changes like this, despite the license. But I''ve the vibe that you're OK with this so I will proceed as `kiwix-zim-updater` and let you know if there are any issues or further questions. Thanks again :)

**Reply by jojo2357** [reply]
@legoktm [here](https://gist.github.com/jojo2357/d056ba908a74ee26fdaea64ffadc5644) ya go! its probably trash, but it is fun. It cant handle `~/` to denote user home, but it is smart enough to only autocomplete directories and once you specify one, it will not continue to suggest them. also has the valid 2 letter country codes baked in. let me know if you need comments on the code.

**Comment by DocDrydenn** [comment]
The script has a built-in updater which looks for the hidden `.git` file within the script's working directory... which is why it's recommended to be installed using a `git clone`. 

That said, if the script isn't installed via `git clone` (or the `.git` file isn't found), then the script should just skip the update check and continue on. (can't remember if it gives a notification when the update is skipped... it should just continue to work as expected either way)

A file name change usually isn't something that's going to upset someone... it's when that name change is done with other changes like changing/removing/hiding original credits, links, sources, repos, etc... People tend to get upset when their work is "stolen".

@jojo2357 I see no real problem with a simple changing of the script's name for the repo, if you see fit to make that change. As for my 2cents... It might, however, cause issues with anyone that already has and uses the script in it's current state. Might have to test out what would happen to those cases? This might also be able to be an easy fix via code with a properly timed update... I have an idea on how it could be done, so if interested, feel free to hit me up. If not, no biggie. Great work on this script. I knew I picked the right person. LOL

**Reply by jojo2357** [reply]
> It might, however, cause issues with anyone that already has and uses the script in it's current state

Yes, right now what would happen is a git pull, then it would fail to restart itself. Also, anyone with CRON going would break if it updated silently.

The only solution I can think of is to rename the script, and replace the old name with a passthrough script to the new one to ensure future compat.

**Comment by jojo2357** [comment]
@legoktm New update 3.4 out now. This fixes critical upstream bugs. No versions prior to 3.4 will work in the future so do with that info what you will
